datasets/signalmixer.py

```diff
@@ -28,109 +28,109 @@ class signalmixer:
-                noise_dataset, batch_size=noise_dataset.dataset_len)))[0]
+        self.snr_range = snr_range
 
-            self.noise_rms = noise_dataset.rms
-
-        self.snr = snr
-        self.noise_kind = noise_kind
-
-        # using getitem to reach the speech test data
-        self.test_dataset_float = next(iter(torch.utils.data.DataLoader(
-            signal_dataset, batch_size=signal_dataset.data.shape[0])))[0]
-
-        if noise_kind == 'WhiteNoise':
-            self.mixed_signal = self.white_noise_mixer()
+        if apply_prob > 1:
+            print('Noise will be applied to the whole dataset! \n')
+            self.apply_prob = 1
         else:
-            self.mixed_signal = self.snr_mixer()
+            self.apply_prob = apply_prob
+        self.noise_type = noise_type
 
     def __getitem__(self, index):
+        inp, target = self.signal_dataset[index]
+        data_type = self.data_type[index]
 
-        inp = self.mixed_signal[index].type(torch.FloatTensor)
-        target = int(self.signal_targets[index])
-        return inp, target
+        # applying noise to train set using per-snr
+        if data_type == 0:
+            if np.random.uniform(0, 1) < self.apply_prob:
+                if self.noise_type is not None:
+                    if not torch.all(inp == 0):
+                        inp = self.snr_mixer(inp)
+
+        # applying noise to all of the test set
+        else:
+            if 'WhiteNoise' in self.noise_type:
+                if not torch.all(inp == 0):
+                    inp = self.white_noise_mixer(inp)
+            elif self.noise_type is not None:
+                if not torch.all(inp == 0):
+                    inp = self.snr_mixer(inp)
+        return inp.type(torch.FloatTensor), int(target)
 
     def __len__(self):
-        return len(self.mixed_signal)
+        return len(self.data)
 
-    def snr_mixer(self):
-        ''' creates mixed signal dataset using the SNR level and the noise dataset
+    def snr_mixer(self, signal):
         '''
-        clean = self.test_dataset_float
-        noise = self.noise_dataset_float
-
-        idx = np.random.randint(0, noise.shape[0], clean.shape[0])
-        noise = noise[idx]
-        rms_noise = self.noise_rms[idx]
+        Creates mixed signals using an SNR level and the noise dataset
+        '''
+        idx_noise = np.random.randint(0, len(self.data))
+        noise, _ = self.noise_dataset[idx_noise]
+        rms_noise = self.noise_dataset.rms[idx_noise]
 
-        snr = self.snr
+        if self.snr_range[0] == self.snr_range[-1]:
+            snr = self.snr_range[0]
+        else:
+            snr = np.random.uniform(self.snr_range[0], self.snr_range[-1])
+        snr = torch.tensor(snr)
 
-        rmsclean = torch.sqrt(torch.mean(clean.reshape(
-            clean.shape[0], -1)**2, 1, keepdims=True)).unsqueeze(1)
+        rmsclean = torch.sqrt(torch.mean(signal**2))
```

Comment:
Please check the variable names like follows:
rmsclean -> rms_clean
scalarclean -> scalar_clean